### ISSUE TITLE
Coroutine type inferred incorrectly

### ISSUE THREAD

**Issue by jolaf** [PyCQA/pylint#4908]
### Bug description

```python
async def f() -> bytes:
    return b''

g = f()
g.send(None)


It looks like type of `g` is inferred as `bytes` instead of "coroutine returning `bytes`".
```

### Command used

```shell
pylint test.py
```

### Pylint output

```shell
************* Module test
test.py:5: [E1101(no-member), ] Instance of 'bytes' has no 'send' member
```

### Expected behavior

No errors.

### Pylint version

```shell
pylint 2.10.2
astroid 2.7.2
Python 3.8.10 (default, Jun  2 2021, 10:49:15) 
[GCC 9.4.0]
```

### OS / Environment

Ubuntu 20.04.3 LTS \n \l

**Comment by nmlorg** [comment]
I just ran into this 🙁.

```
test.py:5:0: E1101: Instance of 'bytes' has no 'send' member (no-member)
```
```
pylint 3.3.1
astroid 3.3.5
Python 3.10.12 (main, Sep 11 2024, 15:47:36) [GCC 11.4.0]
```
